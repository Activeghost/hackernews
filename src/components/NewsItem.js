import React, { Component } from 'react';
import './../App.css';
import PropTypes from "prop-types";


class NewsItem extends Component {
    onDismiss = () => this.props.onClick(this.props.item.objectId);

    render() {
        return <div>
            {
                <div key={this.props.item.objectId}>
                                <span>
                                    <a href={this.props.item.url}>{this.props.item.title}</a>
                                </span>
                    <span>{this.props.item.author}</span>
                    <span>{this.props.item.num_comments}</span>
                    <span>{this.props.item.points}</span>
                    <span>
                            <button
                                onClick={this.onDismiss}
                                type="button">
                                Dismiss
                            </button>
                        </span>
                </div>
            }
        </div>;
    }
}

NewsItem.propTypes = {
    item: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired
};

export default NewsItem;