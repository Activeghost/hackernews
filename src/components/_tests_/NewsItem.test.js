import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NewsItem from './../NewsItem';

const list = [
    {
        title: "React",
        url: "https://facebook.github.io/react",
        author: "Jordan Walke",
        num_comments: 5,
        points: 4,
        objectId: 0
    },
    {
        title: "Redux",
        url: "https://github.com/reactjs/redux",
        author: "Dan Abramov, Andrew Clark",
        num_comments: 2,
        points: 5,
        objectId: 1
    }
];
    test('Renders expected snapshots', () => {
        const newsItem = list[0];
        const result = renderer.create(
            <NewsItem key={newsItem.objectId} item={newsItem} onClick={() => {
            }}/>
        );

        let tree = result.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('calls click event when clicked', () => {
        const newsItem = list[0];
        const onClick = sinon.spy();
        const result = shallow(
            <NewsItem key={newsItem.objectId} item={list[0]} onClick={onClick}/>
        );

       result.find('button').simulate('click');
       expect(onClick.calledWith(0)).toBeTruthy();
    });
