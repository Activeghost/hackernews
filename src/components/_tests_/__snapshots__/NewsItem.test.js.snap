// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders expected snapshots 1`] = `
<div>
  <span>
    <a
      href="https://facebook.github.io/react"
    >
      React
    </a>
  </span>
  <span>
    Jordan Walke
  </span>
  <span>
    5
  </span>
  <span>
    4
  </span>
  <span>
    <button
      onClick={[Function]}
      type="button"
    >
      Dismiss
    </button>
  </span>
</div>
`;
